### PR TITLE
move uncaughtException to main to avoid it report to much times

### PR DESCRIPTION
**Pull request by Jason-Wanxt**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -49,6 +49,18 @@ const main = async () => {
     }
 }
 
+!oneOff && process.on("uncaughtException", async function(e) {
+    if(action === "report") {
+        log(`Uncaught exception in ${argv.chainids.join(",")} ${action} process: ${e.toString()}. restarting in ${argv.rebootMinutes}`, 1)
+        setTimeout(syncRetryablesProcess, 1000 * 60 * argv.rebootMinutes);
+    } else if(action === "update" || action === "sync") {
```
Comment (DZGoldman) [inline src/main.ts]: don't we need two separate conditions here? (i.e., `update` should run `updateProcess`)

File: src/main.ts [hunk 2]
```diff
@@ -49,6 +49,18 @@ const main = async () => {
     }
 }
 
+!oneOff && process.on("uncaughtException", async function(e) {
```
Comment (DZGoldman) [inline src/main.ts]: I think maybe we want to keep the uncaughtException event listener for oneOff, just not have it trigger the recursive setTimeout (i.e., only log). The reason I'd removed it is I thought the process would stay running indefinitely but I don't think that's the case (this was my change, not yours, so I can change it on the one-offs branch later if you'd like)

**Review by DZGoldman — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 2 above)
Comment (Jason-Wanxt) [reply src/main.ts]: Yes we can add uncaughtException support for oneOff, i can help directly on this pr.

**Comment by Jason-Wanxt** [comment]
Just found we don't use `timeout` to restart process on arb-monitor, and it seems if we add it to restart process in uncaughtException handler, it will start multiple process @DZGoldman .

**Comment by DZGoldman** [comment]
@Jason-Wanxt for the one-offs, on uncaught exception can we just report (`log`) but not trigger any timeout ?

**Review by DZGoldman — CHANGES_REQUESTED** [review]
(see comment)

**Comment by Jason-Wanxt** [comment]
> @Jason-Wanxt for the one-offs, on uncaught exception can we just report (`log`) but not trigger any timeout ?

Maybe we shouldn't trigger timeout on non one-off neither? Because just tested on my side, because it will trigger one more process when enter that code block.

**Review by DZGoldman — APPROVED** [review]

**Pull request merged**